### PR TITLE
add carousel, image and text templates and tests

### PR DESCRIPTION
**Pull request by helenaohlsen**

### CODE REVIEW

File: twyla/chat/templates/generic.py [hunk 1]
```diff
@@ -36,13 +48,26 @@ class GenericPayload:
     def add(self, *elements: GenericElement) -> None:
         """
         Add a generic template element
-        :param elements:
+        :param buttons:
```
Comment (abn) [inline twyla/chat/templates/generic.py]: Maybe add some description here

File: twyla/chat/templates/generic.py [hunk 2]
```diff
@@ -1,28 +1,40 @@
-from dataclasses import dataclass, field
-from typing import List
+from dataclasses import dataclass, field, InitVar
+from typing import List, Optional
```
Comment (abn) [inline twyla/chat/templates/generic.py]: Might also want to add `__all__` here to help identify expected class usage.

File: twyla/chat/templates/generic.py [hunk 3]
```diff
@@ -36,13 +48,26 @@ class GenericPayload:
...
+    elements: InitVar[List[GenericElement]] = field(default=None)
+
+    def __post_init__(self, elements: List[GenericElement]):
+        if elements:
+            self.add(*elements)
+
+    def add(self, *elements: GenericElement) -> None:
+        """
+        Add a generic template element
+        :param elements:
```
Comment (abn) [inline twyla/chat/templates/generic.py]: Maybe add some description here.

File: twyla/chat/templates/image.py [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+@dataclass
+class ImagePayload:
+    url: str
+    is_reusable: bool = True
+
+
+@dataclass
+class ImageTemplate:
+    payload: ImagePayload
```
Comment (abn) [inline twyla/chat/templates/image.py]: Would be good for url and is_reusable to be InitVars here and for payload to be set only in a post init function.

Comment (helenaohlsen) [reply twyla/chat/templates/image.py]: @abn so there shouldn't be a separate imagePayload class?

Comment (abn) [reply twyla/chat/templates/image.py]: A separate class yes, but not one that is exported so an end user does not need to be aware of it.

**Pull request merged**